### PR TITLE
Trim trailing whitespace before string comparision

### LINKED ISSUE #2645: "123" == "123  " is incorrect
```fortran
program example
print *, "123" == "123  "
end
```
Prints:
```console
$ lfortran a.f90
False
$ gfortran a.f90 && ./a.out 
 T
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Towards #2645.

### CODE REVIEW

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good to me. Thank you so much for this Pranav!

**Comment by Shaikh-Ubaid** [comment]
@Pranavchiku this seems to fix only the compile-time strings. We might need a fix for runtime strings as well.

File: src/libasr/asr_utils.h [hunk 1]
```diff
@@ -2379,6 +2379,14 @@ static inline ASR::ttype_t* duplicate_type_without_dims(Allocator& al, const ASR
     }
 }
 
+inline std::string remove_trailing_white_spaces(std::string str) {
+    int end = str.size() - 1;
+    while (end >= 0 && std::isspace(str[end])) {
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/asr_utils.h:2384]: I think we should remove only the tailing spaces.

We can also use `return str.substr(0, str.find_last_not_of(' ')+1)`, but this is also fine.

**Comment by Thirumalai-Shaktivel** [comment]
Perfect, thank you! @Pranavchiku do you want to implement this for runtime string comparison as well?

I wanted this PR changes for https://github.com/lfortran/lfortran/pull/2632, if you complete it let me know.

**Comment by Pranavchiku** [comment]
Let's implement for runtime strings in another PR. is it good to merge?

**Comment by Pranavchiku** [comment]
> For runtime you can use builtin trim function, if possible.

LLVM has it. I was thinking about the same.

**Review by certik — APPROVED** [review]
Fine with me. Thanks!

@Thirumalai-Shaktivel is this ok? If so, go ahead and merge.

**Comment by Thirumalai-Shaktivel** [comment]
LGTM, thanks @Pranavchiku!
I will work on the runtime comparison.

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Pull request merged**